### DISCUSSION TITLE
"failed to update config" when saving admin settings

### DISCUSSION THREAD

**Discussion by Qhilm** [immich-app/immich#13395] [Q&A]
The settings are successfully saved actually, so it's just cosmetic, but it's 100% reproducible for me when I change an admin setting, such as job concurrency for example. I'm on v117.0.

The error is a red pop up on the top right, which says: "failed to update config - immich server error".

Any idea what might be causing this?

**Comment by bo0tzz** [comment]
Check the server logs for the actual error

**Comment by Qhilm** [comment]
```
[Nest] 17  - 10/13/2024, 11:11:25 AM     LOG [Api:SmartInfoService~5dmmuklg] Updating database CLIP dimension size to 1024.
[Nest] 17  - 10/13/2024, 11:11:26 AM   ERROR [Api:ErrorInterceptor~5dmmuklg] Database error: QueryFailedError: relation "clip_index" does not exist
```

also happens if I change the logging level for example.

**Comment by Qhilm** [comment]
ok, something was/is seriously wrong, after a power outage, the `immich_server` container doesn't even start anymore:

```
microservices worker error: QueryFailedError: relation "clip_index" does not exist
microservices worker exited with code 1
```

**Comment by mertalev** [comment]
I'm guessing that you restored from a backup recently? If there's a string encoding issue, the index sometimes doesn't get recreated correctly. You can run these SQL commands to fix this:
```sql
SET vectors.pgvector_compatibility=on;
CREATE INDEX IF NOT EXISTS clip_index ON smart_search
USING hnsw (embedding vector_cosine_ops)
WITH (ef_construction = 300, m = 16);
```
And if you have the same issue for `face_index`:
```sql
SET vectors.pgvector_compatibility=on;
CREATE INDEX IF NOT EXISTS face_index ON face_search
USING hnsw (embedding vector_cosine_ops)
WITH (ef_construction = 300, m = 16);
```

**Comment by Qhilm** [comment]
Thanks for getting back to me, I did not restore from a backup at all. What I recently did is moving to a new CLIP model and use remote machine learning. Not sure if it is related.

I was reviewing [this thread](https://github.com/immich-app/immich/issues/11801#issuecomment-2294840717) when you replied.

I ran your SQL commands, but the output is:

```
ERROR:  pgvecto.rs: The given vector is invalid for input.
ADVICE: Check if dimensions and scalar type of the vector is matched with the index.
```

I'm using CLIP `ViT-H-14-quickgelu__dfn5b`.

**Reply by Qhilm** [reply]
I noticed this error in the logs, a bit earlier

```
[Nest] 6  - 10/14/2024, 3:37:19 PM     LOG [Microservices:SmartInfoService] Dimension size of model ViT-H-14-quickgelu__dfn5b is 1024, but database expects 1152.
[Nest] 6  - 10/14/2024, 3:37:19 PM     LOG [Microservices:SmartInfoService] Updating database CLIP dimension size to 1024.
```

hence decided to run:

```
immich=# ALTER TABLE smart_search ALTER COLUMN embedding SET DATA TYPE vector(1024);
ALTER TABLE
immich=# 
```
after that your commands worked (took a while):
```
immich=# SET vectors.pgvector_compatibility=on;
CREATE INDEX IF NOT EXISTS clip_index ON smart_search
USING hnsw (embedding vector_cosine_ops)
WITH (ef_construction = 300, m = 16);
SET
CREATE INDEX
immich=# 
```

Question: should I run Smart Search job on all the assets again?

**Reply by mertalev** [reply]
> Question: should I run Smart Search job on all the assets again?

Yes, that would be a good idea.
